### ISSUE TITLE
MQTT Data Not Displayed in Dashboard Despite Successful Publishing

### ISSUE THREAD

**Issue by ayushGit29** [thingsboard/thingsboard#10995]
**Description:**
We are experiencing an issue with MQTT data ingestion on our ThingsBoard Community Edition instance. Although messages are successfully published and received by an MQTT client, the data does not appear on the ThingsBoard dashboard.

**Environment:**
- ThingsBoard Version: [Specify the version]
- Installation Method: Docker
- MQTT Broker: Embedded within ThingsBoard

**Device Access Token:** `accesstoken`
**MQTT Topic:** `v1/devices/me/telemetry`
**MQTT Client:** Mosquitto (Command Line)

**Steps to Reproduce:**
1. Publish Data via Mosquitto:
   ```sh
   mosquitto_pub -d -q 1 -h localhost -p 1883 -t v1/devices/me/telemetry -u "accesstoken" -m "{temperature:25}"
   
  Result: Message published successfully with debug output confirming PUBACK.

![Screenshot (90)](https://github.com/thingsboard/thingsboard/assets/141584634/f5b705c3-4f3f-498d-9880-5b0bdfd664ef)


![Screenshot (91)](https://github.com/thingsboard/thingsboard/assets/141584634/bcc3871e-9372-4f4f-b208-e5df5ddde06f)

Despite successfully publishing messages to the MQTT broker and verifying their delivery by subscribing to the topic, the data is not appearing on the ThingsBoard dashboard. We have confirmed that the messages are being published and received correctly via MQTT, as we can subscribe to the topic and see the data being transmitted. This indicates that the MQTT setup is functioning as expected. However, ThingsBoard does not seem to process or display the incoming telemetry data. We would greatly appreciate your assistance in diagnosing and resolving this issue to ensure that the telemetry data is correctly ingested and displayed on the ThingsBoard platform.

![Screenshot (93)](https://github.com/thingsboard/thingsboard/assets/141584634/dc7b7f46-60a7-4254-8631-fc3aa4a91de8)

![Screenshot (92)](https://github.com/thingsboard/thingsboard/assets/141584634/07948e20-3f7f-4ad4-b69e-21b3aadadeb5)

**Observations:**

* Messages are published and received correctly via MQTT.
* No telemetry data appears on ThingsBoard.
* No relevant errors in ThingsBoard logs related to MQTT data ingestion.

**Comment by Backdraft007** [comment]
Which rule chain do you use for this device?
Show your rule chain, please.

**Comment by Backdraft007** [comment]
Please enable debug log on "device profile" and "Save timeseries" nodes and look if you can see the incoming messages.

**Comment by ayushGit29** [comment]
I enabled the debug mode in both the "device profile node" and "save timeseries" node:

![Screenshot (95)](https://github.com/thingsboard/thingsboard/assets/141584634/ac62c18d-135d-4f05-8135-29988ddab587)


![Screenshot (96)](https://github.com/thingsboard/thingsboard/assets/141584634/0425f408-769b-4a93-ac31-066d56f1965b)

After this I tried sending the data again using the mosquitto_pub command but it didn't show me any latest telemetry and the device is still inactive.

The "Save Timeseries Node" has no incoming debugging events and the "Device Profile" node has the profile periodic self message.

![Screenshot (100)](https://github.com/thingsboard/thingsboard/assets/141584634/95b0d321-6a07-4a95-a6aa-f54bf5878251)

![Screenshot (101)](https://github.com/thingsboard/thingsboard/assets/141584634/aad149cc-1930-4b99-baa6-ea29f15c021c)

![Screenshot (97)](https://github.com/thingsboard/thingsboard/assets/141584634/b8e1e9d4-946a-4528-ac10-31d225a8cf50)

But the mqtt data is still not displayed in the dashboard despite successfully publishing it.

@Backdraft007 did I do the exact things that you wanted me to do or I did something wrong?

Feel free to follow up and ask any other questions if required.

**Comment by Backdraft007** [comment]
Can you show the used device profile (Details and transport configuration).

**Comment by ayushGit29** [comment]
Here it is. I have used the default profile device. I haven't changed anything in that yet:

![Screenshot (103)](https://github.com/thingsboard/thingsboard/assets/141584634/5b9fb12a-d3c0-4ac9-9fc8-bbc34e129d09)

![Screenshot (104)](https://github.com/thingsboard/thingsboard/assets/141584634/9bd6df0e-e895-43d6-9b7b-f1f3d775a9cd)

![Screenshot (105)](https://github.com/thingsboard/thingsboard/assets/141584634/bb5501a8-1292-4236-872e-35ffd5dbaf86)

I've done nothing after installation. Just created the device and tried sending data through mqtt and it didn't work. I also tried installing the thingsboard community edition in different laptop using the same docker installation but the exact same issue arose there too. 

@Backdraft007

**Comment by Backdraft007** [comment]
Can you post the logs during you send with mqtt?
`docker compose logs -f mytb`

**Comment by myroslav-tb** [comment]
Hi @ayushGit29, From your screenshot, I can see that the device profile rule node does not have any incoming messages from the mosuitto_pub. This means that Thingsboard has not received any messages. The IN messages are generated by the node itself.
![image](https://github.com/thingsboard/thingsboard/assets/158559147/ccc70e83-cc47-45f8-b1d1-54c81874fbd4)

I think another broker is running at port 1883 on your machine. Please try to use port 1884 with mosquitto_pub or turn off another broker.

**Comment by ayushGit29** [comment]
Here is what the output is when I executed the command: 
[Docker Compose Logs.txt](https://github.com/user-attachments/files/15835369/Docker.Compose.Logs.txt)

Here are some of the snapshots:
![Screenshot (108)](https://github.com/thingsboard/thingsboard/assets/141584634/ff045a18-8986-4ed7-b1b1-ec37eefc82c7)

![Screenshot (109)](https://github.com/thingsboard/thingsboard/assets/141584634/9f8bdea0-08f9-47f2-95bc-fc234f177c10)

![Screenshot (110)](https://github.com/thingsboard/thingsboard/assets/141584634/5b073d21-932d-498c-9da6-d59855248867)

![Screenshot (111)](https://github.com/thingsboard/thingsboard/assets/141584634/f853e043-819e-4ae4-b2d0-ca7933f2d944)

![Screenshot (112)](https://github.com/thingsboard/thingsboard/assets/141584634/d8418e50-2c7a-4fff-8ea8-c36bfd46968d)

Hope this might help because I can't really understand anything from it. @Backdraft007

**Comment by ayushGit29** [comment]
@myroslav-tb I think you are right but the docker is running at the 1883 port and I'm running the thingsboard in docker only.

![Screenshot (113)](https://github.com/thingsboard/thingsboard/assets/141584634/e30ad1d8-d2e8-4512-a026-6f1485c6e4f0)

![Screenshot (114)](https://github.com/thingsboard/thingsboard/assets/141584634/38d6bf4a-42bb-42e2-9bbc-634b7a3c4fda)

What should I do?

**Comment by ayushGit29** [comment]
Oh, it actually worked, I just ended the mosquitto.exe task that was running and tried sending the data again and it worked. 

I think it was due to another broker running at 1883 port. 

For anyone who is wondering how the problem got solved, just check the if there are 2 or more brokers running at the same port 1883 and close all the other brokers except the one where thingsboard is running.

To check the brokers running at port 1883 using the following commands in windows:
`netstat -ano | findstr :1883`

If it shows more than one PID (11332 and 3572 in my case),
* go to the task manager ( `ctrl + shift + esc` in windows) 
* click on the details tab and search for the PID's that you found earlier
* close all the other PID's except for the one where the thingsboard is running.

This should solve the problem if you have 2 or more mqtt brokers running at the same port 1883.

Thanks to @Backdraft007 and @myroslav-tb their taking your valuable time to solve this problem which we were trying to solve since a week. Thank you so much for helping, it really means alot. You guys are awesome.

Regards
@ayushGit29
